src/fusion_algorithm.cpp

```diff
@@ -0,0 +1,325 @@
+
+/** @file fusion_algorithm.cpp
+ *  @brief contains the implementation of the sensor fusion algorithm
+ */
+
+// All C++ Headers
+#include "../inc/sensor_fusion_common.hpp"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// All C Headers go here
+
+#ifdef __cplusplus
+}
+#endif
+
+SensorsList_t remove_sensor_by_name(SensorsList_t sensors, char name[SENSOR_MAX_NAME_LEN]) {
+    for (int i = 0; i < sensors.size(); ++i) {
+        if (strcmp(sensors[i].name, name) == 0) {
+            sensors.erase(sensors.begin() + i);
+        }
+    }
+    return sensors;
+}
+
+void delete_element_from_double_array(double *array, int index, int size) {
+    for (int i = index; i < size - 1; ++i) {
+        array[i] = array[i + 1];
+    }
+}
+
+double *get_degree_matrix(SensorsList_t sensors_list) {
```

Comment:
again, we are passing a copy of the sensors list. very heavy. A pointer to that list should be passed.